File: frontend/src/interfaces/Thread.ts
// src/interfaces/Thread.ts
import type { User } from "./User";
import type { Game } from "./Game";
import type { Comment } from "./Comment";

export interface Thread {
  ID: number;
  title: string;
  content: string;
  user_id: number;
  user?: User;
  game_id: number;
  game?: Game;
  comments?: Comment[];
}

export interface CreateThreadRequest {
  title: string;
  content: string;
  user_id: number;
  game_id: number;
}

export interface UpdateThreadRequest {
  ID: number;
  title?: string;
  content?: string;
  user_id?: number;
  game_id?: number;
}

File: backend/entity/thread.go
package entity

import (
	"gorm.io/gorm"
)


type Thread struct {
	gorm.Model
	Title    string `json:"title"`
	Content  string `json:"content"`
	UserID   uint   `json:"user_id"`
	User     *User  `gorm:"foreignKey:UserID" json:"user"`
	GameID   uint   `json:"game_id"`
	Game     *Game  `gorm:"foreignKey:GameID" json:"game"`

	Comments []Comment `gorm:"foreignKey:ThreadID" json:"comments,omitempty"`
}